extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/CommandSpec.scala

```diff
@@ -108,3 +109,19 @@ case class ScanSpec(
     }
   }
 }
+
+case class FunctionSpec(classname: String, functionDescs: Seq[FunctionDesc]) extends CommandSpec {
+  override def opType: String = OperationType.QUERY.toString
+
+  def functions: (Expression) => Seq[Function] = (expr) => {
```

Comment:
After a second thought, I would prefer to extract the functions by adding this method to the ScanSpec without bringing a new Spec in this PR. ScanSpec is solely responsible for resource extracting from logical plans and should be able to cover this scenario.

also cc @yaooqinn